src/api/app/services/rpmlint_log_extractor.rb

```diff
@@ -0,0 +1,56 @@
+class RpmlintLogExtractor
+  attr_reader :project, :package, :repository, :architecture, :log_content
+
+  def initialize(attributes = {})
+    @project = attributes[:project]
+    @package = attributes[:package]
+    @repository = attributes[:repository]
+    @architecture = attributes[:architecture]
+
+    @log_content = ''
+  end
+
+  def call
+    # Retrieve the content of the 'rpmlint.log' file
+    Backend::Api::BuildResults::Binaries.rpmlint_log(project, package, repository, architecture)
+  rescue Backend::NotFoundError => e
+    # TODO: Remove this `unless` condition once request_show_redesign is rolled out
+    return unless Flipper.enabled?(:request_show_redesign, User.session)
+
+    # Case of removed project, package, repository or architecture:
+    return unless e.summary.include?('rpmlint.log: No such file or directory')
```

Comment:
@eduardoj so incase we cannot find the rpmlint.log file and rescue, will this message not always be included in the summary? I havent tested it myself yet, first wanted to ask you

Reply:
ah sorry, its `unless` get it now :)

Reply:
@krauselukas, this backend call could return with a 404 also when the project, package, repository or architecture don't exist. That's why I put the comment just before:
```
# Case of removed project, package, repository or architecture:
```
In those cases, no, the message returned is not related to the `rpmlint.log` file.